### PR TITLE
Add matcher to check if file exists

### PR DESCRIPTION
**Pull request by dg-ratiodata**
This adds a helper to check if a file exists in filesystem. This also resolves the full path. ~~This PR needs to be rebased if #224 is merged (absolute_path -> expand_path).~~

I added a separate spec file. This spec file uses the aruba api as well. To make the "aruba api object" `@aruba` re-usable I added a shared context for that.

### CODE REVIEW

**Comment by dg-ratiodata** [comment]
@jarl-dk @mattwynne Suggestions?

**Comment by dg-ratiodata** [comment]
Mmmh... There are more commits here now than I expected, when I started improving this PR. Hope this is not too much. Otherwise I will try to split it up.

File: spec/support/shared_contexts/aruba.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      include Aruba::Api
+
+      def set_tag(tag_name, value)
+        self.instance_variable_set "@#{tag_name}", value
+      end
+
+      def announce_or_puts(*args)
+        p caller[0..2]
+      end
+    }
```
Comment (mattwynne) [inline spec/support/shared_contexts/aruba.rb]: This feels a bit messy. Is it necessary to define a new instance of this `klass` per example? Could you not just define an `ArubaRunner` class or somesuch once, then use instances of that for `@aruba`?

File: spec/support/shared_contexts/aruba.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    @aruba = klass.new
```
Comment (mattwynne) [inline spec/support/shared_contexts/aruba.rb]: I would prefer to use a `let` block to define `aruba` then use that (without the `@`) in the specs that depend on it.

**Comment by mattwynne** [comment]
Thanks @dg-ratiodata, the new helpers and matchers look great!

File: spec/support/shared_contexts/aruba.rb (hunk 2 above)
Comment (maxmeyer) [inline spec/support/shared_contexts/aruba.rb]: Agreed. I would prefer that too. Is it ok, if I open a new PR for that? I think we need to update the rest of the test suite for that.

File: spec/support/shared_contexts/aruba.rb (hunk 1 above)
Comment (maxmeyer) [inline spec/support/shared_contexts/aruba.rb]: Mmmh... Maybe this should be done in another PR. I only move the (old) code.

Comment (mattwynne) [inline spec/support/shared_contexts/aruba.rb]: Oh I'm sorry I didn't realise! Just go ahead then, and yes we can do that in another PR.

File: spec/support/shared_contexts/aruba.rb (hunk 2 above)
Comment (mattwynne) [inline spec/support/shared_contexts/aruba.rb]: Yes please!

**Comment by maxmeyer** [comment]
Alreday merged by #238
